### PR TITLE
refs #12490 - changes needed for hammer cli

### PR DESCRIPTION
**Pull request by stbenjam**
Some API tweaks for the hammer cli to work well.  Goes with https://github.com/theforeman/hammer_cli_foreman_remote_execution/pull/1.

 :heavy_check_mark: TODO
- [x] Template Input API
- [x] Invocation output API
- [x] template_id => job_template_id in params for hammer id resolution
- [x] RABL clean-up
- [x] Additional API test coverage
- [x] Permissions

### CODE REVIEW

File: app/controllers/api/v2/template_inputs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+      api :GET, "/templates/:template_id/template_inputs/:id", N_("Show template input details")
+      param :template_id, :identifier, :required => true
+      param :id, :identifier, :required => true
+      def show
+      end
+
+      def_param_group :template_input do
+        param :template_input, Hash, :required => true, :action_aware => true do
+          param :name, String, :required => true, :desc => N_('Template name')
```
Comment (iNecas) [inline app/controllers/api/v2/template_inputs_controller.rb]: `s/Template name/Input name/`

File: app/controllers/api/v2/template_inputs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          param :input_type, String, :required => true, :desc => N_('Input type')
```
Comment (iNecas) [inline app/controllers/api/v2/template_inputs_controller.rb]: Enumerating the possible values for the type would make the hammer more usable

File: app/controllers/api/v2/template_inputs_controller.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          param :fact_name, String, :required => false, :desc => N_('Fact name')
```
Comment (iNecas) [inline app/controllers/api/v2/template_inputs_controller.rb]: The help text could include info on what types of input is it valid for

File: app/controllers/api/v2/template_inputs_controller.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      api :GET, "/templates/:template_id/template_inputs/:id", N_("Show template input details")
+      param :template_id, :identifier, :required => true
+      param :id, :identifier, :required => true
+      def show
+      end
+
+      def_param_group :template_input do
+        param :template_input, Hash, :required => true, :action_aware => true do
+          param :name, String, :required => true, :desc => N_('Input name')
+          param :input_type, TemplateInput::TYPES.keys.map(&:to_s), :required => true, :desc => N_('Input type')
```
Comment (stbenjam) [inline app/controllers/api/v2/template_inputs_controller.rb]: Shows the possible types in the docs now

File: app/controllers/api/v2/template_inputs_controller.rb [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+      api :GET, "/templates/:template_id/template_inputs/:id", N_("Show template input details")
+      param :template_id, :identifier, :required => true
+      param :id, :identifier, :required => true
+      def show
+      end
+
+      def_param_group :template_input do
+        param :template_input, Hash, :required => true, :action_aware => true do
+          param :name, String, :required => true, :desc => N_('Input name')
```
Comment (stbenjam) [inline app/controllers/api/v2/template_inputs_controller.rb]: Fixed the name

**Comment by stbenjam** [comment]
Updated, thanks @inecas, good suggestions

File: app/controllers/api/v2/template_inputs_controller.rb (hunk 4 above)
Comment (iNecas) [inline app/controllers/api/v2/template_inputs_controller.rb]: Nice! Could we do the same for `provider_type` on job template?

Comment (stbenjam) [inline app/controllers/api/v2/template_inputs_controller.rb]: Done

**Comment by iNecas** [comment]
ACK, Tested and works well

**Comment by stbenjam** [comment]
Great, thanks!

**Pull request merged**